### ISSUE TITLE
calendar - public caldavs do not work as expected

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roundcubevnz/myroundcube#257]
```
What steps will reproduce the problem?

1. Populate config file like this:

$rcmail_config['public_categories'] = array(
  'Birthdays' => 'f7ff17',
);

$rcmail_config['public_caldavs'] = array(
  'Birthdays' => array(
                   'user' => '%u',
                   'pass' => '%p',
                   'url' => 'https://mycalserver.tld/caldav.php/%u/birthdays/',
                   'auth' => 'basic',
                   'readonly' => false,
                   'extr' => false, // external reminder service (f.e. google calendar reminders)
                 ),
);

  Note that the URL here is fake, it is correct on my system because the $rcmail_config['default_caldav_backend'] has the main calendar on the same URL and it works like a charm.

2. Try to add an event on under the Birthdays category

What is the expected output? What do you see instead?
The expected output should update the calendar with the new event. Instead, 
nothing happens.

What version of the product are you using? On what operating system?

  - Calendar plugin from roundcube-0.6-bundle-v1.7.zip bundle.
  - Davical 0.9.9.7.
  - Roundcube webmail 0.6.
  - Linux, Fedora Core 14.
  - Apache 2.x.

Please provide any additional information below.

The plugin may not be expanding the placeholders as documented on the config 
file, as I see on the apache log files (for the davical virtual host) the 
following:

[05/Nov/2011:01:16:05 +0000] "GET 
/caldav.php/%u/birthdays/20111105T011604Z-8guOhNE HTTP/1.1" 400 316

  Thanks in advance for your time. Congratulations for the excellent plugin.
```

Original issue reported on code.google.com by `cpsoa...@gmail.com` on 5 Nov 2011 at 1:35

**Comment by GoogleCodeExporter** [comment]
```
Please test if the new version works for you:

http://svn.mail4us.net/plugins/calendar.zip
```

Original comment by `rol...@liebl.ath.cx` on 5 Nov 2011 at 9:04
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in Roundcube 0.6 - bundle v1.8
```

Original comment by `rol...@liebl.ath.cx` on 6 Nov 2011 at 6:04
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
  Hi,

  Just tried the last v1.8 bundle and it still doesn't work for me. In the apache log files I still see the %u placeholder not being expanded:

[06/Nov/2011:10:34:40 +0000] "PUT 
/caldav.php/%u/birthdays/20111106T103440Z-16NCwDw HTTP/1.1" 400 316
[06/Nov/2011:10:34:40 +0000] "GET 
/caldav.php/%u/birthdays/20111106T103440Z-16NCwDw HTTP/1.1" 400 316

  Additionaly, when I log in the webmail I get a red notification (on the mail screen on top) stating "Server Error! (Not Found)".
  Thanks for your time.

Regards
```

Original comment by `cpsoa...@gmail.com` on 6 Nov 2011 at 10:38

**Comment by GoogleCodeExporter** [comment]
```
I'm getting the same "Server Error! (Not Found)" error when trying to use Feeds.
```

Original comment by `je.terra...@googlemail.com` on 6 Nov 2011 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
Both things work for me. Is there a chance to get remote access (myroundcube at 
mail4us dot net).
```

Original comment by `rol...@liebl.ath.cx` on 6 Nov 2011 at 4:55
- Changed state: **Re-Opened**

**Comment by GoogleCodeExporter** [comment]
```
  The problem may not be related to feeds as I disabled them to test and I still get the error. Also the "Loading Reminders..." message stays at the bottom of the page forever (maybe that's the reason for the "Server Error!" message?).
  Additionally, I inspected the full apache access log files and those category caldavs are not being authenticated as the accesses are all anonymous, comparing to the default_caldav_backend that is authenticated. So, once again this suggests that all placeholders %u on user and url fields and %p on the pass field of $rcmail_config['public_caldavs'] category array are not being expanded.
```

Original comment by `cpsoa...@gmail.com` on 6 Nov 2011 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
  An update, it seems the "Server Error!" is already tackled down and solution is discussed on Defect 258. It worked for me.
  So, the public_caldavs with expandable placeholders problem is the only one that remains right now.
```

Original comment by `cpsoa...@gmail.com` on 7 Nov 2011 at 12:00

**Comment by GoogleCodeExporter** [comment]
```
Please test new version (link see Comment 1)
```

Original comment by `rol...@liebl.ath.cx` on 7 Nov 2011 at 3:59

**Comment by GoogleCodeExporter** [comment]
```
  Hi,

  Just to report that the version currently at http://svn.mail4us.net/plugins/calendar.zip worked for me.

Thanks
```

Original comment by `cpsoa...@gmail.com` on 7 Nov 2011 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
With the new zip from #9 the Server Error message is gone, but I still
have problems seeing "Loading data..." when accessing the Calendar, and
I have millions of entries in the apache access log like this:

"GET 
/roundcube/?_task=dummy&_task=dummy&_action=plugin.calendar_fetchalllayers&_gmto
ffset=-1&_realgmtoffset=-1&_dststart=1332637200&_dstend=1351386000&_start=132062
0400&_end=1321484400&_category=&_tzname=Europe%2FBerlin&_echo=1&_=1320835768596 
HTTP/1.1" 200 596

It will stop only if I close Roundcube or use the back button to go back to
email.

I can use the E-Mail button in Roundcube only after some minutes when Roundcube
tells me "Request timed out" ...

any idea?

```

Original comment by `haita...@gmail.com` on 9 Nov 2011 at 10:51

**Comment by GoogleCodeExporter** [comment]
```
It looks like some systems have problems with duplicate '_task' param in the 
url.

calendar/program/js/calendar.jsonfeeds.js:

Comment or Remove line 34.
```

Original comment by `rol...@liebl.ath.cx` on 9 Nov 2011 at 4:59

**Comment by GoogleCodeExporter** [comment]
```
does not help :(

Still tons of access.log entries and still 'Loading data ...'
```

Original comment by `haita...@gmail.com` on 10 Nov 2011 at 7:29

**Comment by GoogleCodeExporter** [comment]
```
Could you give me a test account on your system or even better remote access 
(FTP/SSH to Roundcube folder and maybe PHPMyAdmin)?

Details: myroundcube at mail4us dot net
```

Original comment by `rol...@liebl.ath.cx` on 10 Nov 2011 at 8:05

**Comment by GoogleCodeExporter** [comment]
```
Feedback to comment 13 is open. Public CalDAV issue is fixed in 0.6 - bundle 
1.9.
```

Original comment by `rol...@liebl.ath.cx` on 10 Nov 2011 at 12:21

**Comment by GoogleCodeExporter** [comment]
```
@#13: not possible :(
```

Original comment by `haita...@gmail.com` on 10 Nov 2011 at 2:20

**Comment by GoogleCodeExporter** [comment]
```
As I can't reproduce the issue, I have to close the ticket.

Re-open if you have further informations or you are able to reproduce it on my 
demo site http://mail4us.net.
```

Original comment by `rol...@liebl.ath.cx` on 10 Nov 2011 at 2:24
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
:-(
```

Original comment by `haita...@gmail.com` on 11 Nov 2011 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
I've found the problem. The problem is a b0rked jquery-ui in Debian. It claims 
to be 1.8.16 but it is 1.8.15, so I tried the real 1.8.16 and it works like a 
charme :)
```

Original comment by `haita...@gmail.com` on 11 Nov 2011 at 12:06

**Comment by GoogleCodeExporter** [comment]
```
Hmmm ... 

Default Roundcube 0.6 is shipped with jqueryui 1.8.14.

So, why do you have a higher version in jqueryui plugin folder or did you 
override js files included by jqueryui plugin anywhere?
```

Original comment by `rol...@liebl.ath.cx` on 11 Nov 2011 at 1:39

**Comment by GoogleCodeExporter** [comment]
```
I'm using the RoundCube package from Debian, and jqueryui is in the 
roundcube-plugins-extra package and that package depends on libjs-jquery-ui 
package. So the package installs the plugin directory but using symlinks to 
jqueryui package.


```

Original comment by `haita...@gmail.com` on 11 Nov 2011 at 2:09
